src/core/dev_api/openvino/core/so_extension.hpp

```diff
@@ -41,7 +41,11 @@ inline std::string resolve_extension_path(const std::string& path) {
 
 inline std::vector<Extension::Ptr> load_extensions(const std::string& path) {
     const std::string resolved_path = resolve_extension_path(path);
+#ifdef OPENVINO_ENABLE_UNICODE_PATH_SUPPORT
+    auto so = ov::util::load_shared_object(ov::util::string_to_wstring(resolved_path).c_str());
+#else
     auto so = ov::util::load_shared_object(resolved_path.c_str());
```

Comment:
can we do this inside `ov::util::load_shared_object`? It will help us avoid to do the same changes in places using this rountine. How do you think?

Reply:
Yes, we can move it to `ov::util::load_shared_object`. Done.